File: algorithm/HDU-ACM/ambition0109/2111.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 17:04:52
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 2111
////Problem Title: 
////Run result: Accept
////Run time:109MS
////Run memory:6816KB
//////////////////System Comment End//////////////////
import java.util.Arrays;
import java.util.Scanner;

@SuppressWarnings("rawtypes")
class Node implements Comparable {
	int p,m;
	Node(int p,int m){
		this.p=p;
		this.m=m;
	}
	
	@Override
	public int compareTo(Object arg0){
		int p=((Node)arg0).p;
		return p-this.p;
	}
}

public class Main {
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		Node[] node=new Node[105];
		while(true){
			int v=in.nextInt();
			if(v==0) break;
			int n=in.nextInt();
			for(int i=0;i<n;i++){
				int p=in.nextInt();
				int m=in.nextInt();
				node[i]=new Node(p,m);
			}
			Arrays.sort(node,0,n);
			int ans=0,i;
			for(i=0;i<n;i++){
				if(v>node[i].m){
					v-=node[i].m;
					ans+=node[i].p*node[i].m;
				}else break;
			}
			if(i!=n) ans+=node[i].p*v;
			System.out.println(ans);
		}
	}
}

File: algorithm/HDU-ACM/ambition0109/2111.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-05-23 12:29:22
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 2111
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:204KB
//////////////////System Comment End//////////////////
#include<stdio.h>
main()
{
    int v,n,i,j,m[100],p[100],value;
    while(scanf("%d",&v),v)
    {
        for(scanf("%d",&n),i=0;i<n;i++)
            scanf("%d%d",&p[i],&m[i]);
        for(i=0;i<n-1;i++)
            for(j=i+1;j<n;j++)
                if(p[j]>p[i])
                {
                    m[i]^=m[j];m[j]^=m[i];m[i]^=m[j];
                    p[i]^=p[j];p[j]^=p[i];p[i]^=p[j];
                }
        for(i=0,value=0;v-m[i]>=0&&i<n;i++)
        {
            v-=m[i];
            value+=m[i]*p[i];
        }
        if(v&&i!=n) value+=p[i]*v;
        printf("%d\n",value);
    }
}